cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ArtifactHttpHandlerTest.java

```diff
@@ -153,6 +156,29 @@ public void testNotFound() throws IOException, URISyntaxException {
     Assert.assertNull(getArtifact(NamespaceId.DEFAULT.artifact("wordcount", "1.0.0")));
   }
 
+  @Test
+  public void testPluginRequirements() throws Exception {
+    // add a system artifact
+    ArtifactId systemId = NamespaceId.SYSTEM.artifact("wordcount", "1.0.0");
+    addSystemArtifacts();
+
+    Set<ArtifactRange> parents = Sets.newHashSet(new ArtifactRange(
+      systemId.getNamespace(), systemId.getArtifact(),
+      new ArtifactVersion(systemId.getVersion()), true, new ArtifactVersion(systemId.getVersion()), true));
+
+    Manifest manifest = new Manifest();
+    manifest.getMainAttributes().put(ManifestFields.EXPORT_PACKAGE, InspectionApp.class.getPackage().getName());
+    ArtifactId artifactId = NamespaceId.DEFAULT.artifact("inspection", "1.0.0");
+    Assert.assertEquals(HttpResponseStatus.OK.code(),
+                        addPluginArtifact(Id.Artifact.fromEntityId(artifactId), Plugin1.class, manifest, parents)
+                          .getStatusLine().getStatusCode());
+    Set<PluginClass> plugins = getArtifact(artifactId).getClasses().getPlugins();
+    // check that requirement is being stored
+    validateRequirement(plugins, InspectionApp.SingleRequirementPlugin.class.getName(),
+                        ImmutableSet.of(Capabilities.TRANSACTIONS));
+    // For more unit tests see ArtifactInspectorTest#testGetPluginRequirements()
```

Comment:
don't need this comment, it might get renamed or moved or become inaccurate in some other way in the future.